src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/DefaultDocumentWriter.cs

```diff
@@ -70,11 +70,16 @@ public void RenderChildren(RazorIRNode node)
             public override void VisitDocument(DocumentIRNode node)
             {
                 RenderChildren(node);
+
+                Context.Writer.WriteLine("#pragma warning restore 1591");
             }
 
             public override void VisitChecksum(ChecksumIRNode node)
             {
                 Context.BasicWriter.WriteChecksum(Context, node);
+
+                Context.Writer.WriteLine("// <auto-generated/>");
```

Comment:
Having this here is kind of ugly. Spoke to @NTaylorMullen, This is temporary code and is only going to live inside VS and we don't need to care about customization.